tools/run-clangclcheckdocker.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -o noglob
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+MAINSOURCE=$SCRIPTPATH/..
+
+ALL_CROARING_FILES="
```

Comment:
If possible, wouldn't it make more sense to invert this and define the files we _don't_ want to format? That seems easier to keep up to date.

Reply:
@SLieve Can you propose a way to do that?

Reply:
We'd have to generate the list of files dynamically again. Perhaps something like:
```
find $MAINSOURCE -regextype posix-extended -regex '.*\.(h|hh|c|cc|cpp)' | grep -vFf clang-format-ignore.txt
```

Where `clang-format-ignore.txt` contains something like:
```
microbenchmarks/toni_ronnko_dirent.h
microbenchmarks/performancecounters/apple_arm_events.h
...
```

Reply:
Although that would also find files ignored by git. The original approach modified a bit is probably better:
```
git ls-tree --full-tree --name-only -r HEAD | grep -e ".*\.\(c\|h\|cc\|cpp\|hh\)\$" | grep -vFf clang-format-ignore.txt
```

Reply:
Done. Please re-review.